### PR TITLE
ok Travis, smartass

### PR DESCRIPTION
**Pull request by ahjohannessen**
Just me investigating why Travis is failing like a champ.

### CODE REVIEW

File: core/src/main/scala/transport/netty/ClientConnectionPool.scala [hunk 1]
```diff
@@ -251,23 +252,45 @@ class NettyConnectionPool(hosts: Process[Task,InetSocketAddress],
...
+              // add an SSL layer first iff we were constructed with an SslContext, foreach like a unit boss
+
+              sslContext.foreach { s =>
+
+                val sh = s.newHandler(ch.alloc(), addr.getAddress.getHostAddress, addr.getPort)
+
+                sh.handshakeFuture().addListener(new GenericFutureListener[Future[Channel]] {
+                  def operationComplete(future: Future[Channel]): Unit = {
+                    // avoid negotiation when ssl fails
+                    if(!future.isSuccess) { pipe.remove(negotiateCapable) }
```
Comment (ahjohannessen) [inline core/src/main/scala/transport/netty/ClientConnectionPool.scala]: This is brittle and a bit like playing whack-a-mole. There has to be a more robust way to short-circuit.

**Comment by ahjohannessen** [comment]
`ConnectException` happens frequently and that makes me think there is a bug/race around `.start(...).run` because we get connection refused just after having started a server and try to connect to it.

**Comment by timperrett** [comment]
@ahjohannessen is this good to merge? travis passed it, but the comments suggest you're not entirely happy?

**Comment by ahjohannessen** [comment]
@timperrett I would like to test it a bit better. Lately, I've been very busy with work and also been on vacation, but soon I'll put my focus on this again. I've seen some null pointer exceptions when putting client<->server under load and wish to identify root causes and fix that.

**Comment by ahjohannessen** [comment]
Furthermore, it would be interesting to migrate to fs2 in a branch as soon it is ready for the real world :)

**Comment by timperrett** [comment]
Sure, sounds good on both counts :-)

**Comment by ahjohannessen** [comment]
Time to focus on this again, hopefully https://github.com/oncue/remotely/issues/109 can get in first.

**Pull request merged**